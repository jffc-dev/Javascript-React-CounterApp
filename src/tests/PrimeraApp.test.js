import React from 'react';
// import { render } from "@testing-library/react"
import { shallow } from 'enzyme';

import PrimeraApp from "../PrimeraApp"

describe('Pruebas en el componente PrimeraApp', () => {
    // test('debe mostrar el mensaje "Javier Flores"', () => {
        
    //     //Usando test por defecto
    //     // const saludo = "Javier Flores"
    //     // const { getByText } = render( <PrimeraApp saludo={saludo}/>)

    //     // expect(getByText(saludo)).toBeInTheDocument()

        
        
    // })

    test('debe mostrar PrimeraApp, correctamente', () => {
        const saludo = "Javier Flores"
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>)

        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe de mostrar el subtitulo enviado por props', () => {
        const saludo = "Javier Flores"
        const subtitulo = "Desde subtitulos"

        const wrapper = shallow(
        <PrimeraApp 
            saludo={saludo}
            subtitulo={subtitulo}
        />)

        const textParrafo = wrapper.find('p').text()

        expect(textParrafo).toBe(subtitulo)

    })
    
})
