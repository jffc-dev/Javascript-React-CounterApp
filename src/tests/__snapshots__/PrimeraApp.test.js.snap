// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente PrimeraApp debe mostrar PrimeraApp, correctamente 1`] = `
<Fragment>
  <h1>
    Javier Flores
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
